### PR TITLE
feat: added a transaction builder for better argument input.

### PR DESCRIPTION
**Pull request by Foivos**
https://axelarnetwork.atlassian.net/browse/AXE-4143

### CODE REVIEW

**Review by re1ro — APPROVED** [review]

File: scripts/tx-builder.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        let name = `${type.address}::${type.module}::${type.name}`;
+
+        if (type.typeArguments.length > 0) {
+            name += `<${type.typeArguments[0]}`;
+
+            for (let i = 1; i < type.typeArguments.length; i++) {
+                name += `,${type.typeArguments[i]}`;
+            }
+
+            name += '>';
```
Comment (npty) [inline scripts/tx-builder.js:30]: i think this part can be shorten using `join(",")` function e.g. name += `<${type.typeArguments.join(",")}>`

File: scripts/tx-builder.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                package: packageId,
+                module,
+                function: functionName,
+            };
+        }
+
+        const moveFn = await this.client.getNormalizedMoveFunction(target);
+
+        let length = moveFn.parameters.length;
+        if (isTxContext(moveFn.parameters[length - 1])) length = length - 1;
```
Comment (milapsheth) [inline scripts/tx-builder.js:155]: use braces for if blocks

File: scripts/tx-builder.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    if (!parameter) return false;
+    return parameter.address === '0x2' && parameter.module === 'tx_context' && parameter.name === 'TxContext';
+}
+
+function isString(parameter) {
+    if (parameter.MutableReference) parameter = parameter.MutableReference;
+    if (parameter.Reference) parameter = parameter.Reference;
+    parameter = parameter.Struct;
+    if (!parameter) return false;
+    const isAsciiString = parameter.address === '0x1' && parameter.module === 'ascii' && parameter.name === 'String';
```
Comment (milapsheth) [inline scripts/tx-builder.js:126]: add newlines between statements and braces for if blocks

**Review by milapsheth — APPROVED** [review]

File: move/axelar_gateway/info.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {
     "singletons": [
-        "gateway::Gateway",
```
[the remark this answers is not in the document]
Comment (Foivos) [reply move/axelar_gateway/info.json:3]: because it is not created upon deployment, but at a later time. All of them are removed now anyway

**Pull request merged**